javascripts/2048.js

```diff
@@ -1,32 +1,276 @@
 $(document).ready(function() {
   console.log('ready!');
+  arrayInitialize();
+  initializeGame();
   $('body').keydown(function(event){
     var arrow_keys = [37, 38, 39, 40];
+
     if(arrow_keys.indexOf(event.which) > -1) {
-      var tile = $('.tile');
-      moveTile(tile, event.which);
+      // var tile = $('.tile');
+      moveTile(event.which);
       event.preventDefault();
     }
   })
 })
 
-function moveTile(tile, direction) {
+function pop(tile) {
+  $(tile).addClass("popper")
+  .on("animationend",function() {
+    $(this).removeClass("popper");
+  });
+}
+
+
+ //keeps track of where on the board has a tile
+var tile_array = [];
+//set up for 2D array
+function arrayInitialize(){
+  for(var i = 0; i < 4; i++) {
+    var first_array = [];
+    tile_array.push(first_array);
+    for(var j = 0; j < 4; j++) {
+      setUndefinedTile(i, j);
+      }
+    }
+}
+
+function setUndefinedTile(col, row) {
+  tile_array[col][row] = undefined;
```

Comment:
Javascript is a strange beast. `undefined` can actually be redefined to be a different value. WTF right? To be safe, when you want to use undefined, use `void 0`. So in this example:
```
  tile_array[col][row] = void 0;
```